src/components/toolbar.tsx

```diff
@@ -120,6 +120,10 @@ export class ToolbarComponent extends BaseComponent<IProps, IState> {
     const { document: { content } } = this.props;
     const { appConfig: { settings }, ui: { selectedTileIds } } = this.stores;
 
+    const undoManager = this.props.document.treeManagerAPI?.undoManager;
+    if (toolButton.id === "undo" && !undoManager?.canUndo) return true;
+    if (toolButton.id === "redo" && !undoManager?.canRedo) return true;
```

Comment:
@scytacki Same question as above -- do we want clients talking directly to the `UndoManager` or should this be encapsulated at the `DocumentModel` level, for instance?